lib/common/client/middleware/response/facility_parser.rb

```diff
@@ -14,130 +14,164 @@ def on_complete(env)
+            end
 
-          def strip(value)
-            value.respond_to?(:strip) ? value.strip : value
-          end
+            def make_hours_mappings
+              hours_mapping = complex_mapping('hours')
+              hours_mapping.transform_values! do |hours|
+                if /closed/i.match(hours) || hours == '-'
+                  'Closed'
+                else
+                  hours
+                end
+              end
 
-          def services_date(attrs)
-            Date.strptime(attrs['FacilityDataDate'], '%m-%d-%Y').iso8601 if attrs['FacilityDataDate']
-          end
+              hours_mapping
+            end
+
+            def map_health_services
+              {
+                'last_updated' => services_date,
+                'health' => collect_health_services,
+                'other' => []
+              }
+            end
 
-          def services_from_gis(service_map, attrs)
-            return unless service_map
+            def complex_mapping(attr_name)
+              attrs = @src_facility['attributes']
+              item = @mapping[attr_name]
 
-            services = service_map.each_with_object([]) do |(k, v), l|
-              next unless attrs[k] == BaseFacility::YES && BaseFacility::APPROVED_SERVICES.include?(k)
+              return {} unless item
 
-              sl2 = []
-              v.each do |sk|
-                sl2 << sk if attrs[sk] == BaseFacility::YES && BaseFacility::APPROVED_SERVICES.include?(sk)
+              item.each_with_object({}) do |(key, value), hash|
+                hash[key] = value.respond_to?(:call) ? value.call(attrs) : strip(attrs[value])
               end
-              l << { 'sl1' => [k], 'sl2' => sl2 }
             end
-            services.concat(services_from_wait_time_data(attrs['StationNumber'].upcase))
-          end
 
-          def other_services(attrs)
-            services = []
-            services << 'Online Scheduling' if attrs['DirectPatientSchedulingFlag'] == BaseFacility::YES
-            services
-          end
+            def clean_benefits(benefits_hash)
+              benefits_hash.keys.select { |key| benefits_hash[key] == BaseFacility::YES }
+            end
 
-          def services_from_wait_time_data(facility_id)
-            facility_wait_time = FacilityWaitTime.find(facility_id)
-            metric_keys = facility_wait_time&.metrics&.keys || []
-            services = []
-            services << { 'sl1' => ['EmergencyCare'], 'sl2' => [] } if facility_wait_time&.emergency_care&.any?
-            services << { 'sl1' => ['UrgentCare'], 'sl2' => [] } if facility_wait_time&.urgent_care&.any?
-            (Facilities::AccessDataDownload::WT_KEY_MAP.values - %w[primary_care mental_health]).each do |service|
-              services << { 'sl1' => [service.camelize], 'sl2' => [] } if metric_keys.include?(service)
+            def strip(value)
+              value.respond_to?(:strip) ? value.strip : value
```

Comment:
Nitpicky, but I think this (and the few other similar conditionals (e.g. line 128 above)) is a good use case for `try`.
Thoughts?
```suggestion
              value.try(:strip) || value
```

Reply:
`try` is not appropriate here. Depending on the type of object (for example, a Hash), `value` may or may not respond to `try`.

Reply:
Interesting - shouldn't every object respond to `#try`?

Reply:
Yup, you're right. I had tested it in ruby and not rails. Didn't realize it was part of the rails framework